### ISSUE TITLE
如何建模synapse的脉冲是兴奋型还是抑制型？

### ISSUE THREAD

**Issue by tiffinyk** [fangwei123456/spikingjelly#341]
# **Read before creating a new issue**

- Users who want to use SpikingJelly should first be familiar with the usage of PyTorch. 
- If you do not know much about PyTorch, we recommend that the user can learn the [basic tutorials of PyTorch](https://pytorch.org/tutorials/).
- Do not ask for help with the basic conception of PyTorch/Machine Learning but not related to SpikingJelly. For these questions, please refer to Google or [PyTorch Forums](https://discuss.pytorch.org/).

**For faster response**

You can @ the corresponding developers for your issue. Here is the division:

| Features                         | Developers                                                   |
| -------------------------------- | ------------------------------------------------------------ |
| Neurons and Surrogate Functions  | [fangwei123456](https://github.com/fangwei123456)<br />[Yanqi-Chen](https://github.com/Yanqi-Chen) |
| CUDA Acceleration                | [fangwei123456](https://github.com/fangwei123456)<br />[Yanqi-Chen](https://github.com/Yanqi-Chen) |
| Reinforcement Learning           | [lucifer2859](https://github.com/lucifer2859)                |
| ANN to SNN Conversion            | [DingJianhao](https://github.com/DingJianhao)<br />[Lyu6PosHao](https://github.com/Lyu6PosHao) |
| Biological Learning (e.g., STDP) | [AllenYolk](https://github.com/AllenYolk)                    |
| Others                           | [Grasshlw](https://github.com/Grasshlw)<br />[lucifer2859](https://github.com/lucifer2859)<br />[AllenYolk](https://github.com/AllenYolk)<br />[Lyu6PosHao](https://github.com/Lyu6PosHao)<br />[DingJianhao](https://github.com/DingJianhao)<br />[Yanqi-Chen](https://github.com/Yanqi-Chen)<br />[fangwei123456](https://github.com/fangwei123456) |

We are glad to add new developers who are volunteering to help solve issues to the above table.

**Issue type**

- [ ] Bug Report
- [ ] Feature Request
- [ ] Help wanted
- [x] Other

**SpikingJelly version**

`0.0.0.0.2`

**Description**

...

**Minimal code to reproduce the error/bug**

```python
import spikingjelly
# ...
```

**Comment by fangwei123456** [comment]
在SJ框架中，对整个网络层是不做区分的。具体到某个突触（或者说连接），看其符号就行了

**Comment by AllenYolk** [comment]
SpikingJelly 不区分"兴奋性突触"和"抑制性突触"——这一区分由突触权重的符号承担：

- **正权重**：兴奋性（pre spike 增加 post 膜电位）
- **负权重**：抑制性（pre spike 降低 post 膜电位）

实现上 `nn.Linear` 的 `weight` 直接允许任意符号的实数，因此只要 `weight[i, j]` 为负，连接 `j → i` 就是抑制性的。

如果想做"硬约束"（不允许权重变号），可在训练后用 `with torch.no_grad(): fc.weight.clamp_(min=0)` 之类的截断；或自行实现一个 `ConstrainedLinear`，把负权重反向梯度置零。

参考：https://github.com/fangwei123456/spikingjelly/issues/320

**Comment by AllenYolk** [comment]
SpikingJelly does not distinguish "excitatory" vs "inhibitory" synapses — that distinction is carried by the **sign of the weight**:

- **Positive weight** → excitatory (pre-synaptic spike increases post-synaptic membrane potential).
- **Negative weight** → inhibitory (pre-synaptic spike decreases post-synaptic membrane potential).

`nn.Linear`'s `weight` accepts arbitrary real values, so a negative `weight[i, j]` already makes the connection `j → i` inhibitory.

If you want a hard constraint (no sign flips), apply it after training:

```python
with torch.no_grad():
    fc.weight.clamp_(min=0)
```

Or implement a custom `ConstrainedLinear` that zeros the gradient on negative weights.

Reference: https://github.com/fangwei123456/spikingjelly/issues/320
